src/MongoDB/Manager.c

```diff
@@ -212,31 +212,25 @@ PHP_METHOD(Manager, getServers)
 		return;
 	}
 
-	array_init(return_value);
-	set = intern->client->topology->description.servers;
-	for(i=0; i<set->items_len; i++) {
+	sds = mongoc_client_get_server_descriptions(intern->client, &n);
+	array_init_size(return_value, n);
+
+	for (i = 0; i < n; i++) {
 #if PHP_VERSION_ID >= 70000
 		zval obj;
-#else
-		zval *obj = NULL;
-#endif
-
-		mongoc_server_description_t *sd = (mongoc_server_description_t *) set->items[i].item;
 
-		if (sd->type == MONGOC_SERVER_UNKNOWN) {
-			continue;
-		}
-
-#if PHP_VERSION_ID >= 70000
-		phongo_server_init(&obj, intern->client, ((mongoc_server_description_t *)set->items[i].item)->id TSRMLS_CC);
+		phongo_server_init(&obj, intern->client, mongoc_server_description_id(sds[i]) TSRMLS_CC);
 		add_next_index_zval(return_value, &obj);
 #else
-		MAKE_STD_ZVAL(obj);
+		zval *obj = NULL;
 
-		phongo_server_init(obj, intern->client, sd->id TSRMLS_CC);
+		MAKE_STD_ZVAL(obj);
+		phongo_server_init(obj, intern->client, mongoc_server_description_id(sds[i]) TSRMLS_CC);
```

Comment:
`mongoc-server-description.h` only defines:

```
typedef struct _mongoc_server_description_t mongoc_server_description_t;
```

`_mongoc_server_description_t` is defined in `mongoc-server-description-private.h`.